### DISCUSSION TITLE
Add "Texture Interpolation" support for meshes

### DISCUSSION THREAD

**Discussion by IceLancet13** [godot-extended-libraries/godot-ideas#28] [Ideas]
Hi, I'm working on making a game for fun and I'm learning about gridmaps, meshes, and textures. I was using Blender to create a flat plane mesh for the ground, and found that when you load an image texture onto the mesh which is rather small (64x64 pixels), the image gets enlarged causing the image on the mesh to become blurred. Blender has this nice feature called texture interpolation, where you select "Closest" in the drop-down box and it'll make the color gradation of the pixels sharp again. Here's what it looks like with the Linear option:

![2021-11-14 (2)](https://user-images.githubusercontent.com/94293054/141679098-8aa5bfc2-cb93-4b3e-9373-6e9650ccc864.png)

And here's what it looks like with the closest option:

![2021-11-14 (3)](https://user-images.githubusercontent.com/94293054/141679111-4b0f0e67-b974-4200-bf49-9499109b79c6.png)

However when you import the resulting mesh into godot, it becomes blurred again, i.e. it doesn't retain the texture interpolation information from Blender. Here's what the the 2 tiles look like side by side in Godot:

![2021-11-14 (5)](https://user-images.githubusercontent.com/94293054/141679251-ac8a8a2b-36d0-4509-8487-2ce70ab4ec08.png)

I read in the docs that Godot is supposed to fully support importing of meshes from Blender, but in this case it seems like you lose information. If I'm missing something, please let me know. I think having the lines be sharp might be better, though I guess it could be less energy conserving. Also I realized while I was writing this that I could just use a bigger pixel map for the image texture and that ought to resolve the problem, so this may not be necessary. However it could still be useful if it's less graphically demanding than using a higher resolution for the pixel map. I don't know which is better since I don't know much about how it works. But anyway, I will try using a higher resolution, see if that makes it look a bit nicer...

**Comment by Calinou** [comment]
This is already reported on the main issue tracker: https://github.com/godotengine/godot/issues/45206

This is implemented by https://github.com/godotengine/godot/pull/54044, but it's still in draft status as of writing.
